### ISSUE TITLE
Compilation error with react-scripts start: Maximum call stack size exceeded  TSINTERNAL

### ISSUE THREAD

**Issue by zirkelc** [TypeStrong/fork-ts-checker-webpack-plugin#628]
## Current behavior
When I start my react app with `npm run start` it reports the following compilation error:

![image](https://user-images.githubusercontent.com/950244/124376946-a9209280-dca9-11eb-8bb3-b919f6320ffe.png)

## Environment
 - **fork-ts-checker-webpack-plugin**: "4.1.6" (installed by create react app)
 - **typescript**: "^4.1.3"
 - **eslint**: "^7.29.0"
 - **react-scripts**: "4.0.3"
 - **webpack**: "4.44.2" (installed by create react app)
 - **os**: macOS Big Sur 11.3.1

**Comment by piotr-oles** [comment]
Have you tried running `tsc --watch --noEmit`? Does it work?

**Comment by zirkelc** [comment]
No, I was able to solve the issue somehow. 
Typescript or Eslint was reporting a very strange error with a type mismatch from a react hook.

I guess that caused some problems for the type checker but was not reported properly due to react-scripts I guess.
I'm sure running typescript compiler myself without relying on webpack and react-scripts would have highlighted the issue.

Can be closed for now! :)

**Comment by aboutaaron** [comment]
@chriszirkel can you provide any more context on how you solved this issue? I'm getting the exact same error with the same setup as you but not entirely sure where to debug. Any guidance would be helpful!

**Comment by zirkelc** [comment]
@aboutaaron I'm not entirely sure how I solved it. I think I was running ESLint manually from the shell and found a typescript file (a React Hook) that had some TypeScript type problems. When I removed this TypeScript issue, the error was gone.

I think a good starting point is to run the TypeScript compiler and ESLint separately from react-scripts/webpack. I guess `tsc` should report the error properly. 

Let me know if that helps! :)

**Comment by kyleBeikirch** [comment]
I have the same issue only the problem when running typecheck I also get an error 
``` 
<redacted>/node_modules/typescript/lib/tsc.js:56961
            type = eslintgetNarrowableTypeForReference(type, node, checkMode);
            ^
ReferenceError: eslintgetNarrowableTypeForReference is not defined
    at checkIdentifier (<redacted>/node_modules/typescript/lib/tsc.js:56961:13)
    at checkExpressionWorker (<redacted>/node_modules/typescript/lib/tsc.js:63702:28)
    at checkExpression (<redacted>/node_modules/typescript/lib/tsc.js:63657:38)
    at checkNonNullExpression (<redacted>/node_modules/typescript/lib/tsc.js:59167:37)
    at checkPropertyAccessExpression (<redacted>/node_modules/typescript/lib/tsc.js:59212:85)
    at checkExpressionWorker (<redacted>/node_modules/typescript/lib/tsc.js:63734:28)
    at checkExpression (<redacted>/node_modules/typescript/lib/tsc.js:63657:38)
    at checkNonNullExpression (<redacted>/node_modules/typescript/lib/tsc.js:59167:37)
    at checkPropertyAccessExpression (<redacted>/node_modules/typescript/lib/tsc.js:59212:85)
    at checkExpressionWorker (<redacted>/node_modules/typescript/lib/tsc.js:63734:28)
error Command failed with exit code 1.

```

**Comment by cristianghita24** [comment]
@zirkelc any chance you could provide the eslint config/tsconfig that would help finding the issue?

**Comment by zirkelc** [comment]
> @zirkelc any chance you could provide the eslint config/tsconfig that would help finding the issue?

@cristianghita24 it had nothing to do with tsconfig or eslint. The issue was related to a TS error that caused an recursion.

I found the error by running `tsc --noEmit` on the entire project.
